### ISSUE TITLE
Monorepo with pnpm and Vite — 404 errors

### ISSUE THREAD

**Issue by dkzlv** [seek-oss/vanilla-extract#689]
### Describe the bug

I have two packages: `ui` and `app`. `app` imports styled components from `ui`, but the styles do not load resulting in 404 errors for `<component name>.css.ts.vanilla.` files. 

I've seen other issues mentioning the same problem ([1](https://github.com/seek-oss/vanilla-extract/issues/320), [2](https://github.com/seek-oss/vanilla-extract/issues/559)), all are closed and it seems the fix was pushed to a newer version of vite plugin. I currently use the latest one, 3.1.6, and strangely it doesn't work for me.

### Link to reproduction

[Here's a repo](https://github.com/dkzlv/vanilla-extract-repro).

Run `pnpm i && pnpm dev` from root and visit `http://localhost:3000`. Nothing will show:

![image](https://user-images.githubusercontent.com/2846746/168471155-f6adb406-0598-482d-8b6c-33be138cf3a9.png)

### System Info

```node
  System:
    OS: macOS 12.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 2.71 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.17 - /opt/homebrew/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Brave Browser: 101.1.38.111
    Firefox: 99.0.1
    Safari: 15.2
  npmPackages:
    @vanilla-extract/css: ^1.7.0 => 1.7.0 
    @vanilla-extract/vite-plugin: ^3.1.6 => 3.1.6 
    vite: ^2.9.9 => 2.9.9 
```

Also (if it matters):

```
➜  vanilla-extract-repro git:(main) pnpm -v 
7.1.0
```

**Comment by riccardoperra** [comment]
I have the same issue for [codeimage](https://github.com/riccardoperra/codeimage). I had to rollback to `3.1.4` since is the latest version working with a monorepo and `preserveModules: true`

**Comment by jakst** [comment]
I've got the same issues with yarn v3 going from v3.1.4 to v3.1.6. Seems like v3.1.5 is the culprit.

**Comment by mattcompiles** [comment]
Thanks for the clear reproduction. PR to fix this is open #716
